server/app/services/geo/esri/RealEsriClient.java

```diff
@@ -99,23 +112,117 @@ private CompletionStage<WSResponse> tryRequest(WSRequest request, int retries) {
+   * we have results that are greater than the scoring threashold.
+   *
+   * @param addressJson Uncorrected address
+   * @param urls A list of urls that will be processed in order; may stop early if surpassing score
+   *     threshold. The first url in the list will be fetched. Recursive calls will always send in
+   *     all elements except the first one. Processing ends when there are no urls in the list.
+   * @param optionalPreviousRootNode The response from the previous findAddressCandidates call, if
+   *     any
+   * @return The CompletionStage for result of findAddressCandidates, if any. The `candidates` array
+   *     will be the merged results of multiple endpoints if more than one is called.
+   */
+  private CompletionStage<Optional<JsonNode>> processAddressSuggestionUrlsSequentially(
+      ObjectNode addressJson,
+      ImmutableList<String> urls,
+      Optional<JsonNode> optionalPreviousRootNode) {
+    // Base case: All URLs processed or no valid URLs left
+    if (urls.isEmpty()) {
+      return CompletableFuture.completedFuture(optionalPreviousRootNode);
+    }
+
+    // Urls will have at least one item in the list at this point
+    var request = createWebRequest(urls.stream().findFirst().get(), addressJson);
+
+    // Perform the request and handle response
+    return tryRequest(request, this.ESRI_EXTERNAL_CALL_TRIES)
+        .thenCompose(
+            response -> {
+              ESRI_REQUEST_C0UNT.labels(String.valueOf(response.getStatus())).inc();
+              // Skip the first url which we've just called, send the rest to the next recursive
+              // call
+              var nextSetOfUrls = urls.stream().skip(1).collect(ImmutableList.toImmutableList());
+
+              if (response.getStatus() != 200) {
+                // If request fails, proceed to the next URL
+                return processAddressSuggestionUrlsSequentially(
+                    addressJson, nextSetOfUrls, Optional.empty());
+              } else {
+                // Process the successful response
+                JsonNode rootNode = response.asJson();
+
+                // If we have data from a previous call we'll stitch the candidate records together
+                if (optionalPreviousRootNode.isPresent()
+                    && hasCandidatesArray(optionalPreviousRootNode.get())
+                    && hasCandidatesArray(rootNode)) {
+                  optionalPreviousRootNode
+                      .get()
+                      .get(CANDIDATES_NODE_NAME)
+                      .forEach(
+                          candidateNode ->
+                              ((ArrayNode) rootNode.get(CANDIDATES_NODE_NAME)).add(candidateNode));
+                }
+
+                // This will check that all results are under the score threshold. Meaning there
+                // are no result of a high even quality we want to expand to the next url to
```

Comment:
suggestion: This second sentence of this comment is confusing to me. Can you re-word it?

Reply:
Done